packages/components/backtop/src/use-backtop.ts

```diff
@@ -13,12 +13,40 @@ export const useBackTop = (
   const container = shallowRef<Document | HTMLElement>()
   const visible = ref(false)
 
+  const cubic = (value: number): number => value ** 3
+  const easeInOutCubic = (value: number): number =>
+    value < 0.5 ? cubic(value * 2) / 2 : 1 - cubic((1 - value) * 2) / 2
+  const rAF = (callback: any) => {
+    if (typeof window !== 'undefined' && 'requestAnimationFrame' in window) {
+      return window.requestAnimationFrame(callback)
+    } else {
+      return +setTimeout(callback, 16)
+    }
+  }
+  const scrollToTop = () => {
+    if (!el.value) return
+    const beginTime = Date.now()
+    const beginValue = el.value.scrollTop
+    const frameFunc = () => {
+      if (!el.value) return
+      const progress = (Date.now() - beginTime) / 500
+      if (progress < 1) {
+        el.value.scrollTop = beginValue * (1 - easeInOutCubic(progress))
+        rAF(frameFunc)
+      } else {
+        el.value.scrollTop = 0
+      }
+    }
+    rAF(frameFunc)
+  }
+
   const handleScroll = () => {
     if (el.value) visible.value = el.value.scrollTop >= props.visibilityHeight
   }
 
   const handleClick = (event: MouseEvent) => {
-    el.value?.scrollTo({ top: 0, behavior: 'smooth' })
+    // el.value?.scrollTo({ top: 0, behavior: 'smooth' })
+    el.value && scrollToTop()
```

Comment:
`scrollToTop` also has judgment `el.value`, you can directly execute `scrollToTop`.

Reply:
You're right!